### PR TITLE
Implement Confiant callback for blocked ads

### PR DESCRIPTION
**Pull request by buck06191**
## What does this change?
When Confiant blocks an ad we want to recover lost revenue by refreshing the ad slot at most once to find an alternative solution.

Co-authored-by: Ioanna Kyprianou <ioanna.kyprianou.contractor@guardian.co.uk>
Co-authored-by: Max Duval <max@mxdvl.com>

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?
Will allow us to refresh a blocked ad slot, reducing lost revenue. Success should be reflected in increased earnings. This may require a long AB test in order to determine if this is successful.

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/ad-verification/prepare-ad-verification.js [hunk 1]
```diff
@@ -8,14 +8,66 @@ const errorHandler = (error: Error) => {
...
+    console.log("*** Has slot already been refreshed? ", confiantRefreshedSlots.includes(blockedSlotPath))
+    // check if ad is blocked and haven't refreshed the slot yet.
+    if (isBlocked && !confiantRefreshedSlots.includes(blockedSlotPath) ) {
+        console.log('*** Ad slot blocked');
+        const slots = window.googletag.pubads().getSlots();
+        slots.forEach((currentSlot) => {
+            if (blockedSlotPath === currentSlot.getSlotElementId()) {
+                console.log('*** Refreshing blocked ad slot ', currentSlot);
+                // refresh the blocked slot to get new ad
+                window.googletag.pubads().refresh([currentSlot]);
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/ad-verification/prepare-ad-verification.js:50]: I think this should be calling [refreshAdvert] or [loadAdvert], do in [/dfp/load-advert.js][2]

[refreshAdvert]:https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/commercial/modules/dfp/load-advert.js#L50
[loadAdvert]:https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/commercial/modules/dfp/load-advert.js#L32

[2]: https://github.com/guardian/frontend/tree/master/static/src/javascripts/projects/commercial/modules/dfp/load-advert.js

That is what [displayAds] is calling

[displayAds]: 
https://github.com/guardian/frontend/tree/master/static/src/javascripts/projects/commercial/modules/dfp/display-ads.js#L7

**Comment by buck06191** [comment]
This needs AB testing adding in still before merging

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala [hunk 2]
```diff
@@ -318,6 +318,64 @@ object DataModelV3 {
 
 object DotcomponentsDataModel {
   val VERSION = 2
```
Comment (ioanna0) [inline article/app/model/dotcomponents/DotcomponentsDataModel.scala:320]: what are these server side changes ? 🤔

Comment (buck06191) [reply article/app/model/dotcomponents/DotcomponentsDataModel.scala:320]: It looks like a rebase has added in some file changes?

**Comment by buck06191** [comment]
Closing this due to a bad rebase. Carried over to https://github.com/guardian/frontend/pull/22724

**Pull request closed without merging**